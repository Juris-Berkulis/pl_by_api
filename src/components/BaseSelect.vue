<script setup lang="ts">
import type { SelectedProjectStatus } from '@/types';

interface Props {
    selectedValue: SelectedProjectStatus,
    optionsList: Array<SelectedProjectStatus>,
};

defineProps<Props>();

defineEmits(['update:selectedValue']);
</script>

<template>
<select 
    class="border p-1 focus:outline-none"
    :value="selectedValue" 
    @change="$emit('update:selectedValue', ($event.target as HTMLSelectElement).value as SelectedProjectStatus)"
>
    <option v-for="option of optionsList" :key="option">{{ option }}</option>
</select>
</template>

<style scoped>
</style>
