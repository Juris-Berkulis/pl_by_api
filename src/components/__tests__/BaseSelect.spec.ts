import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import BaseSelect from '../BaseSelect.vue'


describe('BaseSelect', () => {
    const wrapper = mount(BaseSelect, { 
        props: { 
            selectedValue: 'Любой', 
            optionsList: ['Любой', 'CREATED', 'IN_PROGRESS', 'FINISHED'] 
        } 
    });

    it('Snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Renders properly', () => {
        expect(wrapper.text()).toContain('Любой');
    });

    it('Contains text', () => {
        const wrapper = mount(BaseSelect, {
            props: { 
                selectedValue: 'CREATED', 
                optionsList: ['Любой', 'CREATED', 'IN_PROGRESS', 'FINISHED'] 
            } 
        });
        expect(wrapper.get('select').attributes('value') === 'CREATED');
        expect(wrapper.text()).toContain('Любой');
        expect(wrapper.text()).toContain('CREATED');
        expect(wrapper.text()).toContain('IN_PROGRESS');
        expect(wrapper.text()).toContain('FINISHED');
    });
});
