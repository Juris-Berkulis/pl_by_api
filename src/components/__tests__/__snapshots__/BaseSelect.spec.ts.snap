// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseSelect > Snapshot 1`] = `
"<select class=\\"w-full sm:w-1/3 border p-1 text-sm focus:outline-none\\">
  <option>Любой</option>
  <option>CREATED</option>
  <option>IN_PROGRESS</option>
  <option>FINISHED</option>
</select>"
`;
